networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/libp2p/LibP2PPrivateKeyLoader.java

```diff
@@ -41,7 +42,12 @@ public PrivKey get() {
         privateKeySource
             .map(PrivateKeySource::getOrGeneratePrivateKeyBytes)
             .orElseGet(() -> generateAndSaveNewPrivateKey(keyValueStore));
-    return KeyKt.unmarshalPrivateKey(privKeyBytes.toArrayUnsafe());
+    try {
+      return KeyKt.unmarshalPrivateKey(privKeyBytes.toArrayUnsafe());
+    } catch (final Exception __) {
+      // Try to unmarshall raw SECP256K1 key without extra ProtoBuf data
+      return Secp256k1Kt.unmarshalSecp256k1PrivateKey(privKeyBytes.toArrayUnsafe());
```

Comment:
I think it would be better to catch exception even here and throw a new exception reporting the 2 attempts. WDYT?